### PR TITLE
Index distributions

### PR DESCRIPTION
**Pull request by fgravin**
I did a mapping to match dcat (esri, ods) format to

- link protocols
- mimetype

I now index

- format
- mediaType

From ODS, in the index

```json
{
 "protocol": "WWW:DOWNLOAD",
 "name": "csv Format",
 "format": "csv",
 "description": "- *Geo Point* : geo_point_2d[geo_point_2d]\\n- *Geo Shape* : geo_shape[geo_shape]\\n- *osm_id* : osm_id[text]\\n- *type* : type[text]\\n- *Nom* : name[text]\\n- *Marque* : brand[text]\\n- *Opérateur* : operator[text]\\n- *network* : network[text]\\n- *wheelchair* : wheelchair[text]\\n- *opening_hours* : opening_hours[text]\\n- *bank_id_code* : bank_id_code[text]\\n- *Distributeur* : has_atm[text]\\n- *brand_wikidata* : brand_wikidata[text]\\n- *com_insee* : com_insee[text]\\n- *Commune* : com_nom[text]\\n",
 "mediaType": "text/csv",
 "url": "https://metropole-europeenne-de-lille.opendatasoft.com/explore/dataset/banques-et-distributeurs-de-billets/download?format=csv&timezone=Europe/Berlin&use_labels_for_header=false"
}
```

From ESRI

```json
[
 {
 "protocol": "WWW:LINK",
 "name": "ArcGIS Hub Dataset",
 "format": "web page",
 "description": "ArcGIS Hub Dataset",
 "mediaType": "text/html",
 "url": "https://data-atmo-hdf.opendata.arcgis.com/datasets/4ee9ddcfc11c4d0cb92d54d825534ff9_0"
 },
 {
 "protocol": "ESRI:REST",
 "name": "ArcGIS GeoService",
 "format": "arcgis geoservices rest api",
 "description": "ArcGIS GeoService",
 "mediaType": "application/json",
 "url": "https://services8.arcgis.com/rxZzohbySMKHTNcy/arcgis/rest/services/mes_hdf_journalier_poll_princ/FeatureServer/0"
 },
 {
 "protocol": "WWW:DOWNLOAD",
 "name": "GeoJSON",
 "format": "geojson",
 "description": "GeoJSON",
 "mediaType": "application/vnd.geo+json",
 "url": "https://data-atmo-hdf.opendata.arcgis.com/datasets/4ee9ddcfc11c4d0cb92d54d825534ff9_0.geojson?outSR=%7B%22latestWkid%22%3A3857%2C%22wkid%22%3A102100%7D"
 },
 {
 "protocol": "WWW:DOWNLOAD",
 "name": "CSV",
 "format": "csv",
 "description": "CSV",
 "mediaType": "text/csv",
 "url": "https://data-atmo-hdf.opendata.arcgis.com/datasets/4ee9ddcfc11c4d0cb92d54d825534ff9_0.csv?outSR=%7B%22latestWkid%22%3A3857%2C%22wkid%22%3A102100%7D"
 },
 {
 "protocol": "WWW:DOWNLOAD",
 "name": "KML",
 "format": "kml",
 "description": "KML",
 "mediaType": "application/vnd.google-earth.kml+xml",
 "url": "https://data-atmo-hdf.opendata.arcgis.com/datasets/4ee9ddcfc11c4d0cb92d54d825534ff9_0.kml?outSR=%7B%22latestWkid%22%3A3857%2C%22wkid%22%3A102100%7D"
 },
 {
 "protocol": "WWW:DOWNLOAD",
 "name": "Shapefile",
 "format": "zip",
 "description": "Shapefile",
 "mediaType": "application/zip",
 "url": "https://data-atmo-hdf.opendata.arcgis.com/datasets/4ee9ddcfc11c4d0cb92d54d825534ff9_0.zip?outSR=%7B%22latestWkid%22%3A3857%2C%22wkid%22%3A102100%7D"
 }
]
```

### CODE REVIEW

File: src/main/plugin/dcat2/index-fields/index.xsl [hunk 1]
```diff
@@ -42,6 +42,7 @@
 
   <xsl:import href="common/index-utils.xsl"/>
   <xsl:import href="../layout/utility-tpl-multilingual.xsl"/>
+  <xsl:import href="../convert/distrubution-mapping.xsl"/>
```
Comment (jahow) [inline src/main/plugin/dcat2/index-fields/index.xsl]: typo in filename

File: src/main/plugin/dcat2/index-fields/index.xsl [hunk 2]
```diff
@@ -248,13 +249,30 @@
...
+          <xsl:variable name="protocol" select="$format-mimetype-mapping/entry[format=$format]/protocol"/>
+          <xsl:variable name="linkName" select="dct:title"/>
+          <xsl:variable name="description" select="dct:description"/>
+          <xsl:variable name="url" select="dcat:accessURL"/>
+
+          <link type="object">{
+            "protocol":"<xsl:value-of select="gn-fn-index:json-escape($protocol)"/>",
+            "url":"<xsl:value-of select="gn-fn-index:json-escape($url)"/>",
+            "name":"<xsl:value-of select="$linkName"/>",
+            "format":"<xsl:value-of select="$format"/>",
```
Comment (jahow) [inline src/main/plugin/dcat2/index-fields/index.xsl:270]: mediaType is not indexed, is that on purpose? I mean, I think `format` is enough but I wasn't sure when reading the commit message

**Review by jahow — APPROVED** [review]
Not tested but the examples look very good, thanks! I think the `mediaType` indexation is missing though (see remark).

File: src/main/plugin/dcat2/index-fields/index.xsl (hunk 2 above)
Comment (fgravin) [reply src/main/plugin/dcat2/index-fields/index.xsl:270]: Nice catch it's because i'm working in the data_dir and forgot to get it back.
Thanks

**Pull request merged**